amp/WEB-INF/src/org/digijava/kernel/ampapi/endpoints/activity/preview/PreviewActivityService.java

```diff
@@ -0,0 +1,290 @@
+package org.digijava.kernel.ampapi.endpoints.activity.preview;
+
+import static java.util.stream.Collectors.groupingBy;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.core.Response;
+
+import org.dgfoundation.amp.currencyconvertor.AmpCurrencyConvertor;
+import org.digijava.kernel.ampapi.endpoints.activity.InterchangeUtils;
+import org.digijava.kernel.ampapi.endpoints.errors.ApiError;
+import org.digijava.kernel.ampapi.endpoints.errors.ApiRuntimeException;
+import org.digijava.kernel.exception.DgException;
+import org.digijava.module.aim.dbentity.AmpActivityVersion;
+import org.digijava.module.aim.dbentity.AmpCurrency;
+import org.digijava.module.aim.dbentity.AmpFunding;
+import org.digijava.module.aim.dbentity.AmpFundingAmount;
+import org.digijava.module.aim.dbentity.FundingInformationItem;
+import org.digijava.module.aim.helper.Constants;
+import org.digijava.module.aim.util.ActivityUtil;
+import org.digijava.module.aim.util.CurrencyUtil;
+import org.digijava.module.categorymanager.dbentity.AmpCategoryValue;
+import org.digijava.module.categorymanager.util.CategoryConstants;
+import org.digijava.module.common.util.DateTimeUtil;
+
+/**
+ * 
+ * @author Viorel Chihai
+ *
+ */
+public final class PreviewActivityService {
+    
+    protected static final int PERCENTAGE_MULTIPLIER = 100;
+
+    private static PreviewActivityService previewActivityService;
+
+    private PreviewActivityService() {
+    }
+
+    public static PreviewActivityService getInstance() {
+        if (previewActivityService == null) {
+            previewActivityService = new PreviewActivityService();
+        }
+
+        return previewActivityService;
+    }
+
+    public PreviewActivityFunding getPreviewActivityFunding(Long activityId, Long currencyId) {
+        PreviewActivityFunding activityFunding = new PreviewActivityFunding();
+        try {
+            AmpActivityVersion activity = ActivityUtil.loadActivity(activityId);
+            
+            AmpCurrency currency = null;
+            if (currencyId != null) {
+                currency = CurrencyUtil.getAmpcurrency(currencyId);
+            }
+            
+            if (currencyId == null || currency == null) {
+                throw new ApiRuntimeException(Response.Status.BAD_REQUEST, ApiError.toError(
+                        "Parameter 'currency-id' is not specified or it is not a valid value"));
```

Comment:
Please also check input for right `activityId` and define constants to use as a parameter name in EP and here in error details to ensure consistency.